### DISCUSSION TITLE
crop image in UNETR

### DISCUSSION THREAD

**Discussion by ouyuxuanbridge** [Project-MONAI/MONAI#4125] [Q&A]
Hi,

I am a little confused about this "allow smaller" in Monai.transforms, does it mean some of the image,  especially the margin parts  won' t be used in training?  

Besides, in my last attempt of UNETR , I tried not to use the random crop, with a image size of 64*64*64 , will the data be loaded in the model still be of the same size?

I deleted these two: (https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/unetr_btcv_segmentation_3d.ipynb)
CropForegroundd(keys=["image", "label"], source_key="image"),
        RandCropByPosNegLabeld(
            keys=["image", "label"],
            label_key="label",
            spatial_size=(96, 96, 96),
            pos=1,
            neg=1,
            num_samples=4,
            image_key="image",
            image_threshold=0,
        ),

Thanks!

I really appreciate your help!

**Comment by Nic-Ma** [comment]
Hi @Ouyuxuan623 ,

The `allow_smaller` arg means: when computing box size with `margin` to **crop**, whether allow the **image size** to be smaller than **box size**, more details: https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/array.py#L706
If True, the cropping can work on even small images.

About the `UNETR` question, @ahatamiz may help share some comments?

Thanks in advance.

**Comment by ahatamiz** [comment]
If random crop is not used, then the input will be same as the original image size, whether re-sampled or not, which can be problematic for UNETR, since it expects a fixed size resolution due to its patch embedding layer. I recommend using random crop.
